source/nodejs/adaptivecards-designer/src/card-designer.ts

```diff
@@ -317,29 +321,39 @@ export class CardDesigner {
 
             this.buildTreeView();
         };
-        this._designerSurface.onCardValidated = (parseErrors: Array<Adaptive.IValidationError>, validationResults: Adaptive.ValidationResults) => {
+        this._designerSurface.onCardValidated = (logEntries: Adaptive.IValidationEvent[]) => {
+            if (this.onCardValidated) {
+                this.onCardValidated(this, logEntries);
+            }
+
             let errorPane = document.getElementById("errorPane");
             errorPane.innerHTML = "";
 
-            if (parseErrors.length > 0) {
-                let errorMessages: Array<string> = [];
+            if (logEntries.length > 0) {
+                let dedupedEntries: Adaptive.IValidationEvent[] = [];
 
-                for (let error of parseErrors) {
-                    if (errorMessages.indexOf(error.message) < 0) {
-                        errorMessages.push(error.message);
+                for (let entry of logEntries) {
+                    if (dedupedEntries.indexOf(entry) < 0) {
+                        dedupedEntries.push(entry);
                     }
                 }
 
-                for (let message of errorMessages) {
-                    errorPane.appendChild(this.renderErrorPaneElement("[Error] " + message));
-                }
-            }
-
-            if (validationResults.failures.length > 0) {
-                for (let failure of validationResults.failures) {
-                    for (let error of failure.errors) {
-                        errorPane.appendChild(this.renderErrorPaneElement("[" + failure.cardObject.getJsonTypeName() + "] " + error.message, failure.cardObject));
+                for (let entry of dedupedEntries) {
```

Comment:
nit: consider adding line 342-353 at line 337 after removing ln 337

Reply:
I find it more legible to keep as is (e.g. first build deduped array, then loop through it.)